extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js

```diff
@@ -779,6 +786,27 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
     return super.renderNonAmpCreative();
   }
 
+  /** @override */
+  layoutCallback() {
+    return super.layoutCallback().then(superReturn => {
+      // We expand fluid here because we must first wait for the iframe to fire
+      // onload.
+      this.expandFluidCreative_();
+      return superReturn;
+    });
+  }
+
+  /** @override */
+  viewportCallback(inViewport) {
+    if (this.reattemptToExpandFluidCreative_ && !inViewport) {
+      // If the initial expansion attempt failed (e.g., the slot was within the
+      // viewport), then we will re-attempt to expand it here whenever the slot
+      // is outside the viewport.
+      this.expandFluidCreative_();
+    }
+    super.viewportCallback(inViewport);
```

Comment:
Any reason not to make the super call first?

Reply:
Done.